src/VisualStudio/Core/Def/ColorSchemes/ColorSchemeApplier.cs

```diff
@@ -70,25 +70,38 @@ public async Task InitializeAsync(CancellationToken cancellationToken)
             _isInitialized = true;
         }
 
-        // We need to update the theme whenever the Editor Color Scheme setting changes.
-        await TaskScheduler.Default;
-        var settingsManager = await _asyncServiceProvider.GetServiceAsync<SVsSettingsPersistenceManager, ISettingsManager>(_threadingContext.JoinableTaskFactory).ConfigureAwait(false);
-
-        await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
-        settingsManager.GetSubset(ColorSchemeOptionsStorage.ColorSchemeSettingKey).SettingChangedAsync += ColorSchemeChangedAsync;
-
-        await TaskScheduler.Default;
-
-        // Try to migrate the `useEnhancedColorsSetting` to the new `ColorSchemeName` setting.
-        _settings.MigrateToColorSchemeSetting();
+        packageRegistrationTasks.AddTask(isMainThreadTask: false, task: PackageInitializationBgThreadAsync);
+    }
 
-        // Since the Roslyn colors are now defined in the Roslyn repo and no longer applied by the VS pkgdef built from EditorColors.xml,
-        // We attempt to apply a color scheme when the Roslyn package is loaded. This is our chance to update the configuration registry
-        // with the Roslyn colors before they are seen by the user. This is important because the MEF exported Roslyn classification
-        // colors are only applicable to the Blue and Light VS themes.
+    private async Task PackageInitializationBgThreadAsync(IProgress<ServiceProgressData> progress, PackageRegistrationTasks packageRegistrationTasks, CancellationToken cancellationToken)
+    {
+        var settingsManager = await _asyncServiceProvider.GetServiceAsync<SVsSettingsPersistenceManager, ISettingsManager>(_threadingContext.JoinableTaskFactory).ConfigureAwait(false);
 
-        // If the color scheme has updated, apply the scheme.
-        await UpdateColorSchemeAsync(cancellationToken).ConfigureAwait(false);
+        packageRegistrationTasks.AddTask(
+            isMainThreadTask: true,
+            task: (progress, packageRegistrationTasks, cancellationToken) =>
+            {
+                // We need to update the theme whenever the Editor Color Scheme setting changes.
+                settingsManager.GetSubset(ColorSchemeOptionsStorage.ColorSchemeSettingKey).SettingChangedAsync += ColorSchemeChangedAsync;
+
+                packageRegistrationTasks.AddTask(
+                    isMainThreadTask: false,
+                    task: async (progress, packageRegistrationTasks, cancellationToken) =>
+                    {
+                        // Try to migrate the `useEnhancedColorsSetting` to the new `ColorSchemeName` setting.
+                        _settings.MigrateToColorSchemeSetting();
```

Comment:
deos this part actually need to happen during package registration?  or could it be just bg work we fire off and forget about? (i genuinely don't know.  just seeing how to minimize the work duruing package registration).

Reply:
As mentioned offline, I don't know. This PR is focused on creating some infrastructure to allow us to reduce thread switches from various package registration callers. Will dig in more into this specific area later.